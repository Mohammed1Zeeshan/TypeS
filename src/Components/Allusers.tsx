import axios from 'axios';
import React, { useEffect, useState } from 'react'
import { Button, Nav, Table } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import { IUser } from '../Intefaces/IUser';
import agent from '../Service/BaseUrl';
 

//  below is the user properties 

interface Props
{
  users:IUser[];
}

function Allusers()
 {
  const [users,setUsers]=useState<IUser[]>([]);

  useEffect(()=>{loadData();},[])

   const loadData=async ()=>{
     const res= await axios.get('http://localhost:3005/users')
     console.log(res.data);
     setUsers(res.data);
   }  
   const deleteuser=async (id:number)=>{
    await axios.delete(`http://localhost:3005/users/${id}`)
    loadData();
  }

  return (
    <div>
    <Table  striped bordered hover className='table table-borderless'>
            <thead className='thead-dark'>
            <tr>
            <th>KEY</th>
            <th>VALUE</th>
            <th>ACTION</th>
            </tr>
            </thead>
            <tbody>
            {
                users.map(user=>(
                    <tr>
                        <th>{user.Key}</th>
                        <th>{user.Value}</th>
                         <th>
                        <Link to={`/users/${user.id}`}><button className='btn btn-primary'>View</button> </Link>
                        <Button  onClick={()=> deleteuser(user.id)}>Delete</Button>
                         </th>
                    </tr>  
                )) 
            }
            </tbody>
        </Table>      
        </div>
  )
}

export default Allusers