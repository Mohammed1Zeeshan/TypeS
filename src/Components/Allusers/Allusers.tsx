import axios from 'axios';
import React, { useEffect, useState } from 'react'
import { Button, Nav, Table } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import { IUser } from '../../Intefaces/IUser';
 

//  below is the user properties 

interface Props
{
  users:IUser[];
}

function Allusers()
 {
  const [users,setUsers]=useState<IUser[]>([]);

  useEffect(()=>{loadData();},[])

   const loadData=async()=>{
 const res= await axios.get('https://localhost:44330/api/User/users')
     console.log(res.data);
     setUsers(res.data);
     
   }  
   const deleteuser=async (id:number)=>{
    //await axios.delete(`http://localhost:3005/users/${id}`)
    await axios.delete(`https://localhost:44330/api/User/DELETE/${id}`)
    loadData();
  }
//useEffect(()=>{fetch('https://localhost:44330/api/User/users').then(res=>res.json()).then((result)=>{setUsers(result);})},[])
  return (
    <div  className='Alluser'>
       
    <table  className='table border table-striped'>
            <thead className='thead-dark'>
            <tr>
            <th key='key' className='tab'>KEY</th>
            <th key='value' className='tab'>VALUE</th>
            <th key='action' className='tab'>ACTION</th>
            </tr>
            </thead>
            <tbody>
            {
            users.map(user=>(
                    <tr>
                        <th>{user.key}</th>
                        <th>{user.value}</th>
                         <th>
                        <Link to={`/users/${user.id}`}><button className='btn btn-primary'>View</button> </Link>
                        <Button  className='delete' onClick={()=> deleteuser(user.id)}>Delete</Button>
                         </th>
                    </tr>  
                )) 
            }
            </tbody>
        </table> 
        
        </div> 
      
  )
}

export default Allusers