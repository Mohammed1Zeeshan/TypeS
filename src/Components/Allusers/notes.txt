 
// we can use regular id but its bad practise so we use component name as id thats for functionaly of projects
data-testid='componentname'

next we have to use this where we want to render this component 