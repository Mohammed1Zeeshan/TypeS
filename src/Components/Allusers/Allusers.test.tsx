
import Allusers from './Allusers';
import {shallow } from "enzyme";
import { render, screen } from "@testing-library/react";
import { Button } from "react-bootstrap";
 import puppeteer from 'puppeteer-core';
import {toMatchImageSnapshot} from 'jest-image-snapshot';

jest.setTimeout(30000)
expect.extend({toMatchImageSnapshot});
const browserPath="C:\\Program Files (x86)\\Google\\Chrome\\Application\\chrome.exe";

const mockFn = jest.fn();

describe('Testing ',()=>{
    it('Checking for key ',()=>{
      render(<Allusers/>)
      const LinkElement =screen.getByText(/Key/i)
      expect(LinkElement).toBeInTheDocument();
    });
    it('Checking for Value ',()=>{
        render(<Allusers/>)
        const LinkElement =screen.getByText(/Value/i)
        expect(LinkElement).toBeInTheDocument();
      });
      it('Checking for Action ',()=>{
        render(<Allusers/>)
        const LinkElement =screen.getByText(/Action/i)
        expect(LinkElement).toBeInTheDocument();
      });
})
 
describe('shallow', ()=>{
    it('shallow should render the component correctly',()=>{
        const wraper=shallow(<Allusers/>)
        expect(wraper).toMatchSnapshot();
    })
    it('check the length of the column in a table',()=>{
      const wraper=shallow(<Allusers/>)
      expect(wraper.find('.tab').length).toBe(3);
  })
   it('Key as the content',()=>{
    const wraper=shallow(<Allusers/>)
    expect(wraper.find('.tab').at(0).key()).toBe('key');
})
it('Value as the content',()=>{
  const wraper=shallow(<Allusers/>)
  expect(wraper.find('.tab').at(1).key()).toBe('value');
})
it('Value as the content',()=>{
  const wraper=shallow(<Allusers/>)
  expect(wraper.find('.tab').at(2).key()).toBe('action');
})
it('Delete',()=>{
  const wraper=shallow(<Button name='delete' onClick={mockFn}/>)
  expect(mockFn).toMatchSnapshot();
})
})

// test for function exam[ple test


// image snap shot testing
// describe('Imagesanpshot testing', () => {
//   test('Alluser page rendering', async() => { 
//     const browser =await puppeteer.launch();
//     const page  =await browser.newPage();
    
    

    it('create add user snapshot',async()=>{
    const browser = await puppeteer.launch({executablePath:browserPath,});
    const page=await browser.newPage();
    await page.goto('http://localhost:3006');
    const image= await page.screenshot()
    expect(image).toMatchImageSnapshot();
  });


