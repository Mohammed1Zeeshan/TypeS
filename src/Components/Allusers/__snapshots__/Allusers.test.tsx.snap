// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow Delete 1`] = `[MockFunction]`;

exports[`shallow shallow should render the component correctly 1`] = `
<div
  className="Alluser"
>
  <table
    className="table border table-striped"
  >
    <thead
      className="thead-dark"
    >
      <tr>
        <th
          className="tab"
          key="key"
        >
          KEY
        </th>
        <th
          className="tab"
          key="value"
        >
          VALUE
        </th>
        <th
          className="tab"
          key="action"
        >
          ACTION
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
